RequireThisCheck not able to handle try-with-resources

I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#RequireThis
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java
```
```bash
/var/tmp $ cat config.xml
```
```XML
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="RequireThisCheck">
      <property name="validateOnlyOverlapping" value="false" />
    </module>
  </module>
</module>
```
```bash
/var/tmp $ cat Test.java
```
```java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Test {

    public BufferedReader instFld2;

     void method() {
        try (BufferedReader instFld2 = new BufferedReader(new InputStreamReader(null, "utf-8"))) {
        }
        catch (IOException e) {
        }
    }
}
```
```bash
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.3.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:10:29: Reference to instance variable 'instFld2' needs "this.". [RequireThis]
Audit done.
Checkstyle ends with 1 errors.
```

The violation is given for resource definition, which is not expected.

--- a/.ci/pitest-suppressions/pitest-coding-require-this-check-suppressions.xml
+++ b/.ci/pitest-suppressions/pitest-coding-require-this-check-suppressions.xml
@@ -112,8 +112,8 @@
     <sourceFile>RequireThisCheck.java</sourceFile>
     <mutatedClass>com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck</mutatedClass>
     <mutatedMethod>leaveToken</mutatedMethod>
-    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.RemoveSwitchMutator_6</mutator>
-    <description>RemoveSwitch 6 mutation</description>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.RemoveSwitchMutator_9</mutator>
+    <description>RemoveSwitch 9 mutation</description>
     <lineContent>switch (ast.getType()) {</lineContent>
   </mutation>
 

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java
@@ -587,15 +587,7 @@ public class RequireThisCheck extends AbstractCheck {
                 frameStack.addFirst(new BlockFrame(frame, ast));
                 break;
             case TokenTypes.METHOD_DEF:
-                final DetailAST methodFrameNameIdent = ast.findFirstToken(TokenTypes.IDENT);
-                final DetailAST mods = ast.findFirstToken(TokenTypes.MODIFIERS);
-                if (mods.findFirstToken(TokenTypes.LITERAL_STATIC) == null) {
-                    ((ClassFrame) frame).addInstanceMethod(methodFrameNameIdent);
-                }
-                else {
-                    ((ClassFrame) frame).addStaticMethod(methodFrameNameIdent);
-                }
-                frameStack.addFirst(new MethodFrame(frame, methodFrameNameIdent));
+                collectMethodDeclarations(frameStack, ast, frame);
                 break;
             case TokenTypes.CTOR_DEF:
             case TokenTypes.COMPACT_CTOR_DEF:
@@ -652,6 +644,26 @@ public class RequireThisCheck extends AbstractCheck {
         else {
             frame.addIdent(ident);
         }
+    }
+
+    /**
+     * Collects {@code METHOD_DEF} declarations.
+     *
+     * @param frameStack stack containing the FrameTree being built.
+     * @param ast AST to parse.
+     * @param frame current frame.
+     */
+    private static void collectMethodDeclarations(Deque<AbstractFrame> frameStack,
+                                                  DetailAST ast, AbstractFrame frame) {
+        final DetailAST methodFrameNameIdent = ast.findFirstToken(TokenTypes.IDENT);
+        final DetailAST mods = ast.findFirstToken(TokenTypes.MODIFIERS);
+        if (mods.findFirstToken(TokenTypes.LITERAL_STATIC) == null) {
+            ((ClassFrame) frame).addInstanceMethod(methodFrameNameIdent);
+        }
+        else {
+            ((ClassFrame) frame).addStaticMethod(methodFrameNameIdent);
+        }
+        frameStack.addFirst(new MethodFrame(frame, methodFrameNameIdent));
     }
 
     /**

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java
@@ -275,7 +275,8 @@ public class RequireThisCheck extends AbstractCheck {
         TokenTypes.PARAMETER_DEF,
         TokenTypes.TYPE_ARGUMENT,
         TokenTypes.RECORD_DEF,
-        TokenTypes.RECORD_COMPONENT_DEF
+        TokenTypes.RECORD_COMPONENT_DEF,
+        TokenTypes.RESOURCE
     );
     /** Set of all assign tokens. */
     private static final BitSet ASSIGN_TOKENS = TokenUtil.asBitSet(
@@ -362,6 +363,8 @@ public class RequireThisCheck extends AbstractCheck {
             TokenTypes.IDENT,
             TokenTypes.RECORD_DEF,
             TokenTypes.COMPACT_CTOR_DEF,
+            TokenTypes.LITERAL_TRY,
+            TokenTypes.RESOURCE,
         };
     }
 
@@ -406,8 +409,13 @@ public class RequireThisCheck extends AbstractCheck {
             case TokenTypes.RECORD_DEF:
                 current.push(frames.get(ast));
                 break;
+            case TokenTypes.LITERAL_TRY:
+                if (ast.getFirstChild().getType() == TokenTypes.RESOURCE_SPECIFICATION) {
+                    current.push(frames.get(ast));
+                }
+                break;
             default:
-                // do nothing
+                break;
         }
     }
 
@@ -425,8 +433,13 @@ public class RequireThisCheck extends AbstractCheck {
             case TokenTypes.RECORD_DEF:
                 current.pop();
                 break;
+            case TokenTypes.LITERAL_TRY:
+                if (current.peek().getType() == FrameType.TRY_WITH_RESOURCES_FRAME) {
+                    current.pop();
+                }
+                break;
             default:
-                // do nothing
+                break;
         }
     }
 
@@ -556,6 +569,12 @@ public class RequireThisCheck extends AbstractCheck {
                     frame.addIdent(parameterIdent);
                 }
                 break;
+            case TokenTypes.RESOURCE:
+                final DetailAST resourceIdent = ast.findFirstToken(TokenTypes.IDENT);
+                if (resourceIdent != null) {
+                    frame.addIdent(resourceIdent);
+                }
+                break;
             case TokenTypes.CLASS_DEF:
             case TokenTypes.INTERFACE_DEF:
             case TokenTypes.ENUM_DEF:
@@ -599,6 +618,11 @@ public class RequireThisCheck extends AbstractCheck {
                 if (isAnonymousClassDef(ast)) {
                     frameStack.addFirst(new AnonymousClassFrame(frame,
                             ast.getFirstChild().toString()));
+                }
+                break;
+            case TokenTypes.LITERAL_TRY:
+                if (ast.getFirstChild().getType() == TokenTypes.RESOURCE_SPECIFICATION) {
+                    frameStack.addFirst(new TryWithResourcesFrame(frame, ast));
                 }
                 break;
             default:
@@ -653,6 +677,11 @@ public class RequireThisCheck extends AbstractCheck {
                 break;
             case TokenTypes.LITERAL_NEW:
                 if (isAnonymousClassDef(ast)) {
+                    frames.put(ast, frameStack.poll());
+                }
+                break;
+            case TokenTypes.LITERAL_TRY:
+                if (ast.getFirstChild().getType() == TokenTypes.RESOURCE_SPECIFICATION) {
                     frames.put(ast, frameStack.poll());
                 }
                 break;
@@ -1228,6 +1257,8 @@ public class RequireThisCheck extends AbstractCheck {
         CATCH_FRAME,
         /** For frame type. */
         FOR_FRAME,
+        /** Try with resources frame type. */
+        TRY_WITH_RESOURCES_FRAME
 
     }
 
@@ -1676,6 +1707,24 @@ public class RequireThisCheck extends AbstractCheck {
             return FrameType.CATCH_FRAME;
         }
 
+        @Override
+        protected AbstractFrame getIfContains(DetailAST identToFind, boolean lookForMethod) {
+            final AbstractFrame frame;
+
+            if (!lookForMethod
+                    && containsFieldOrVariable(identToFind)) {
+                frame = this;
+            }
+            else if (getParent().getType() == FrameType.TRY_WITH_RESOURCES_FRAME) {
+                // Skip try-with-resources frame because resources cannot be accessed from catch
+                frame = getParent().getParent().getIfContains(identToFind, lookForMethod);
+            }
+            else {
+                frame = getParent().getIfContains(identToFind, lookForMethod);
+            }
+            return frame;
+        }
+
     }
 
     /**
@@ -1696,6 +1745,29 @@ public class RequireThisCheck extends AbstractCheck {
         @Override
         public FrameType getType() {
             return FrameType.FOR_FRAME;
+        }
+
+    }
+
+    /**
+     * A frame initiated on entering a try-with-resources construct;
+     * holds local resources for the try block.
+     */
+    private static class TryWithResourcesFrame extends AbstractFrame {
+
+        /**
+         * Creates try-with-resources frame.
+         *
+         * @param parent parent frame.
+         * @param ident ident frame name ident.
+         */
+        protected TryWithResourcesFrame(AbstractFrame parent, DetailAST ident) {
+            super(parent, ident);
+        }
+
+        @Override
+        public FrameType getType() {
+            return FrameType.TRY_WITH_RESOURCES_FRAME;
         }
 
     }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java
@@ -1346,29 +1346,29 @@ public class RequireThisCheck extends AbstractCheck {
         /**
          * Check whether the frame contains a field or a variable with the given name.
          *
-         * @param nameToFind the IDENT ast of the name we're looking for.
+         * @param identToFind the IDENT ast of the name we're looking for.
          * @return whether it was found.
          */
-        protected boolean containsFieldOrVariable(DetailAST nameToFind) {
-            return containsFieldOrVariableDef(varIdents, nameToFind);
+        protected boolean containsFieldOrVariable(DetailAST identToFind) {
+            return containsFieldOrVariableDef(varIdents, identToFind);
         }
 
         /**
          * Check whether the frame contains a given name.
          *
-         * @param nameToFind IDENT ast of the name we're looking for.
+         * @param identToFind IDENT ast of the name we're looking for.
          * @param lookForMethod whether we are looking for a method name.
          * @return whether it was found.
          */
-        protected AbstractFrame getIfContains(DetailAST nameToFind, boolean lookForMethod) {
+        protected AbstractFrame getIfContains(DetailAST identToFind, boolean lookForMethod) {
             final AbstractFrame frame;
 
             if (!lookForMethod
-                && containsFieldOrVariable(nameToFind)) {
+                && containsFieldOrVariable(identToFind)) {
                 frame = this;
             }
             else {
-                frame = parent.getIfContains(nameToFind, lookForMethod);
+                frame = parent.getIfContains(identToFind, lookForMethod);
             }
             return frame;
         }
@@ -1401,8 +1401,8 @@ public class RequireThisCheck extends AbstractCheck {
          * @return true if ast is correspondent to ident.
          */
         protected boolean isProperDefinition(DetailAST ident, DetailAST ast) {
-            final String nameToFind = ident.getText();
-            return nameToFind.equals(ast.getText())
+            final String identToFind = ident.getText();
+            return identToFind.equals(ast.getText())
                 && CheckUtil.isBeforeInSource(ast, ident);
         }
     }
@@ -1573,27 +1573,27 @@ public class RequireThisCheck extends AbstractCheck {
         }
 
         @Override
-        protected boolean containsFieldOrVariable(DetailAST nameToFind) {
-            return containsFieldOrVariableDef(instanceMembers, nameToFind)
-                    || containsFieldOrVariableDef(staticMembers, nameToFind);
+        protected boolean containsFieldOrVariable(DetailAST identToFind) {
+            return containsFieldOrVariableDef(instanceMembers, identToFind)
+                    || containsFieldOrVariableDef(staticMembers, identToFind);
         }
 
         @Override
         protected boolean isProperDefinition(DetailAST ident, DetailAST ast) {
-            final String nameToFind = ident.getText();
-            return nameToFind.equals(ast.getText());
+            final String identToFind = ident.getText();
+            return identToFind.equals(ast.getText());
         }
 
         @Override
-        protected AbstractFrame getIfContains(DetailAST nameToFind, boolean lookForMethod) {
+        protected AbstractFrame getIfContains(DetailAST identToFind, boolean lookForMethod) {
             AbstractFrame frame = null;
 
-            if (lookForMethod && containsMethod(nameToFind)
-                || containsFieldOrVariable(nameToFind)) {
+            if (lookForMethod && containsMethod(identToFind)
+                || containsFieldOrVariable(identToFind)) {
                 frame = this;
             }
             else if (getParent() != null) {
-                frame = getParent().getIfContains(nameToFind, lookForMethod);
+                frame = getParent().getIfContains(identToFind, lookForMethod);
             }
             return frame;
         }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java
@@ -337,6 +337,31 @@ public class RequireThisCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testTryWithResources() throws Exception {
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWithInlineConfigParser(
+                getPath("InputRequireThisTryWithResources.java"), expected);
+    }
+
+    @Test
+    public void testTryWithResourcesOnlyOverlappingFalse() throws Exception {
+        final String[] expected = {
+            "44:23: " + getCheckMessage(MSG_VARIABLE, "fldCharset", ""),
+            "57:13: " + getCheckMessage(MSG_VARIABLE, "fldCharset", ""),
+            "69:45: " + getCheckMessage(MSG_METHOD, "methodToInvoke", ""),
+            "77:24: " + getCheckMessage(MSG_METHOD, "methodToInvoke", ""),
+            "103:51: " + getCheckMessage(MSG_VARIABLE, "fldBufferedReader", ""),
+            "107:23: " + getCheckMessage(MSG_VARIABLE, "fldBufferedReader", ""),
+            "107:54: " + getCheckMessage(MSG_VARIABLE, "fldScanner", ""),
+            "110:24: " + getCheckMessage(MSG_VARIABLE, "fldStreamReader", ""),
+            "111:23: " + getCheckMessage(MSG_VARIABLE, "fldBufferedReader", ""),
+            "111:54: " + getCheckMessage(MSG_VARIABLE, "fldScanner", ""),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputRequireThisTryWithResourcesOnlyOverlappingFalse.java"), expected);
+    }
+
+    @Test
     public void testCatchVariables() throws Exception {
         final String[] expected = {
             "38:21: " + getCheckMessage(MSG_VARIABLE, "ex", ""),

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisTryWithResources.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisTryWithResources.java
@@ -1,0 +1,114 @@
+/*
+RequireThis
+checkFields = (default)true
+checkMethods = (default)true
+validateOnlyOverlapping = (default)true
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.requirethis;
+
+import java.io.*;
+import java.util.Scanner;
+
+public class InputRequireThisTryWithResources implements AutoCloseable {
+    private BufferedReader fldBufferedReader;
+    private InputStreamReader fldStreamReader;
+    private Scanner fldScanner;
+    private String fldCharset = "utf-8";
+
+    void oneResource() {
+        try (BufferedReader fldBufferedReader
+                     = new BufferedReader(new InputStreamReader(null, "utf-8"))) { } // ok
+        catch (IOException e) { }
+    }
+
+    void twoResourcesReferencingEachOther() {
+        try (InputStreamReader fldStreamReader = new InputStreamReader(null, "utf-8"); // ok
+             BufferedReader fldBufferedReader = new BufferedReader(fldStreamReader)) { } // ok
+        catch (IOException e) { }
+    }
+
+    void threeResourcesReferencingEachOther() {
+        try (InputStreamReader fldStreamReader = new InputStreamReader(null, "utf-8"); // ok
+             BufferedReader fldBufferedReader = new BufferedReader(fldStreamReader); // ok
+             Scanner fldScanner = new Scanner(
+                     fldStreamReader.toString() + fldBufferedReader.toString())) { } // ok
+        catch (IOException e) { }
+    }
+
+    void failToHandleParameter() {
+        try (BufferedReader fldBufferedReader = new BufferedReader(
+                new InputStreamReader(null, fldCharset))) { } // ok
+        catch (IOException e) { }
+    }
+
+    void handleParameter() {
+        try (BufferedReader fldBufferedReader = new BufferedReader(
+                new InputStreamReader(null, this.fldCharset))) { } // ok
+        catch (IOException e) { }
+    }
+
+    void noResources() {
+        try {
+            int a = 5;
+            fldCharset += a; // ok
+        }
+        catch (Exception ex) { }
+    }
+
+    String methodToInvoke() {
+        return "string";
+    }
+
+    void methodInvoke() {
+        try (BufferedReader fldBufferedReader = new BufferedReader(
+                new InputStreamReader(null, methodToInvoke()))) { } // ok
+        catch (Exception ex) { }
+    }
+
+    void methodIdentCopy() {
+        try (BufferedReader methodToInvoke = new BufferedReader(
+                new InputStreamReader(null, this.methodToInvoke()))) { // ok
+            String a = methodToInvoke() + methodToInvoke.toString(); // ok
+        }
+        catch (Exception ex) { }
+    }
+
+    final static InputRequireThisTryWithResources r1 = new InputRequireThisTryWithResources();
+    final InputRequireThisTryWithResources r2 = new InputRequireThisTryWithResources();
+    static InputRequireThisTryWithResources r3 = new InputRequireThisTryWithResources();
+    InputRequireThisTryWithResources r4 = new InputRequireThisTryWithResources();
+
+    void staticVariables() {
+        try (r1) { } // ok
+        catch (Exception e) { }
+
+        try (r1.r2) { } // ok
+        catch (Exception e) { }
+
+        try (r1.r2.r4.r2.r4.r2) { } // ok
+        catch (Exception e) { }
+    }
+
+    void nestedTryWithResources() {
+        try (InputStreamReader fldStreamReader = new InputStreamReader(null, "utf-8")) { // ok
+            try (BufferedReader fldBufferedReader = new BufferedReader(fldStreamReader)) { // ok
+                try (Scanner fldScanner = new Scanner(
+                        fldBufferedReader.toString() + fldStreamReader.toString())) { } // ok
+            }
+            try (Scanner fldScanner = new Scanner(
+                    fldBufferedReader.toString() + fldStreamReader.toString())) { } // ok
+            String a = fldStreamReader.toString() // ok
+                    + fldBufferedReader.toString() + fldScanner.toString(); // ok
+        }
+        catch (IOException e) {
+            String a = fldStreamReader.toString() // ok
+                    + fldBufferedReader.toString() + fldScanner.toString(); // ok
+        }
+    }
+
+    @Override
+    public void close() throws Exception { }
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisTryWithResourcesOnlyOverlappingFalse.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisTryWithResourcesOnlyOverlappingFalse.java
@@ -1,0 +1,117 @@
+/*
+RequireThis
+checkFields = (default)true
+checkMethods = (default)true
+validateOnlyOverlapping = false
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.requirethis;
+
+import java.io.*;
+import java.util.Scanner;
+
+public class InputRequireThisTryWithResourcesOnlyOverlappingFalse implements AutoCloseable {
+    private BufferedReader fldBufferedReader;
+    private InputStreamReader fldStreamReader;
+    private Scanner fldScanner;
+    private String fldCharset = "utf-8";
+
+    void oneResource() {
+        try (BufferedReader fldBufferedReader
+                     = new BufferedReader(new InputStreamReader(null, "utf-8"))) { } // ok
+        catch (IOException e) { }
+    }
+
+    void twoResourcesReferencingEachOther() {
+        try (InputStreamReader fldStreamReader = new InputStreamReader(null, "utf-8"); // ok
+             BufferedReader fldBufferedReader = new BufferedReader(fldStreamReader)) { } // ok
+        catch (IOException e) { }
+    }
+
+    void threeResourcesReferencingEachOther() {
+        try (InputStreamReader fldStreamReader = new InputStreamReader(null, "utf-8"); // ok
+             BufferedReader fldBufferedReader = new BufferedReader(fldStreamReader); // ok
+             Scanner fldScanner
+                     = new Scanner(fldStreamReader.toString()
+                     + fldBufferedReader.toString())) { } // ok
+        catch (IOException e) { }
+    }
+
+    void failToHandleParameter() {
+        try (BufferedReader fldBufferedReader = new BufferedReader(new InputStreamReader(
+                null, fldCharset))) { } // violation '.*variable 'fldCharset'.*'
+        catch (IOException e) { }
+    }
+
+    void handleParameter() {
+        try (BufferedReader fldBufferedReader = new BufferedReader(
+                new InputStreamReader(null, this.fldCharset))) { } // ok
+        catch (IOException e) { }
+    }
+
+    void noResources() {
+        try {
+            int a = 5;
+            fldCharset += a; // violation '.*variable 'fldCharset'.*'
+        }
+        catch (Exception ex) { }
+    }
+
+    String methodToInvoke() {
+        return "string";
+    }
+
+    void methodInvoke() {
+        try (BufferedReader fldBufferedReader = new BufferedReader(
+                // violation below 'Method .* 'methodToInvoke' needs "this."'
+                new InputStreamReader(null, methodToInvoke()))) { }
+        catch (Exception ex) { }
+    }
+
+    void methodIdentCopy() {
+        try (BufferedReader methodToInvoke = new BufferedReader(
+                new InputStreamReader(null, this.methodToInvoke()))) { // ok
+            // violation below 'Method .* 'methodToInvoke' needs "this."'
+            String a = methodToInvoke() + methodToInvoke.toString();
+        }
+        catch (Exception ex) { }
+    }
+
+    final static InputRequireThisTryWithResources r1 = new InputRequireThisTryWithResources();
+    final InputRequireThisTryWithResources r2 = new InputRequireThisTryWithResources();
+    static InputRequireThisTryWithResources r3 = new InputRequireThisTryWithResources();
+    InputRequireThisTryWithResources r4 = new InputRequireThisTryWithResources();
+
+    void staticVariables() {
+        try (r1) { } // ok
+        catch (Exception e) { }
+        try (r1.r2) { } // ok
+        catch (Exception e) { }
+        try (r1.r2.r4.r2.r4.r2) { } // ok
+        catch (Exception e) { }
+    }
+
+    void nestedTryWithResources() {
+        try (InputStreamReader fldStreamReader = new InputStreamReader(null, "utf-8")) { // ok
+            try (BufferedReader fldBufferedReader = new BufferedReader(fldStreamReader)) { // ok
+                try (Scanner fldScanner = new Scanner(
+                        fldBufferedReader.toString() + fldStreamReader.toString())) { } // ok
+            }
+            // violation below '.*variable 'fldBufferedReader' needs "this."'
+            try (Scanner fldScanner = new Scanner(fldBufferedReader.toString()
+                    + fldStreamReader.toString())) { }
+
+            String a = fldStreamReader.toString()
+                    + fldBufferedReader.toString() + fldScanner.toString(); // 2 violations
+        }
+        catch (IOException e) {
+            String a = fldStreamReader.toString() // violation
+                    + fldBufferedReader.toString() + fldScanner.toString(); // 2 violations
+        }
+    }
+
+    @Override
+    public void close() throws Exception { }
+}

